prometheus-client-metrics/src/main/scala/org/http4s/client/prometheus/PrometheusClientMetrics.scala

```diff
@@ -0,0 +1,131 @@
+package org.http4s.client.prometheus
+
+import io.prometheus.client._
+import cats.implicits._
+import cats.data.Kleisli
+import cats.effect.Sync
+import org.http4s._
+import org.http4s.client.{Client, DisposableResponse}
+
+object PrometheusClientMetrics {
+
+  private def reportStatus(status: Status): String =
+    status.code match {
+      case hundreds if hundreds < 200 => "1xx"
+      case twohundreds if twohundreds < 300 => "2xx"
+      case threehundreds if threehundreds < 400 => "3xx"
+      case fourhundreds if fourhundreds < 500 => "4xx"
+      case _ => "5xx"
+    }
+
+  /** The value of this key in the request's attributes is used as the value for the destination metric label. */
+  val Destination = AttributeKey[String]
```

Comment:
I sometimes want to phase out attributes, but I can't get rid of all the use cases for them.  And here we have another.

Alternative: maybe a function `Request => Option[String]`  could be provided, so the destination could be calculated from the request (probably the `Uri`?)  itself.

Reply:
I like the idea of a generic `Request => Option[String]`, will try that out. Maybe even `Request => List[String]` and let user specify a `List[String]` of label names when creating the middleware?